lib/liquid/block_body.rb

```diff
@@ -48,6 +54,17 @@ def parse(tokenizer, parse_context)
       yield nil, nil
     end
 
+    def whitespace_handler(token, parse_context)
+      if token[2] == WhitespaceControl
+        previous_token = @nodelist.pop
```

Comment:
Won't this pop a node from the nodelist even if the previous token wasn't a string?